app/web/eslint.config.js

```diff
@@ -0,0 +1,200 @@
+// import reactRefresh from "eslint-plugin-react-refresh";
+import { FlatCompat } from "@eslint/eslintrc";
+import eslint from "@eslint/js";
+import eslintParser from "@typescript-eslint/parser";
+import importPlugin from "eslint-plugin-import";
+import noRelativeImportPlugin from "eslint-plugin-no-relative-import-paths";
+import unusedImportsPlugin from "eslint-plugin-unused-imports";
+import { dirname } from "path";
+import tseslint from "typescript-eslint";
+import { fileURLToPath } from "url";
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+
+const compat = new FlatCompat({
+  baseDirectory: import.meta.dirname,
+});
+
+const boolPrefixes = ["is", "should", "has", "can", "did", "will"];
+
+const baseNamingConventions = [
+  {
+    selector: "default",
+    format: ["camelCase"],
+  },
+  {
+    selector: "variable",
+    modifiers: ["destructured"],
+    format: null,
+  },
+  {
+    selector: "variable",
+    types: ["boolean"],
+    format: ["PascalCase"],
+    prefix: boolPrefixes,
+  },
+  {
+    selector: "variable",
+    modifiers: ["const", "global"],
+    types: ["boolean"],
+    format: ["UPPER_CASE"],
+    prefix: boolPrefixes.map((prefix) => `${prefix.toUpperCase()}_`),
+  },
+  {
+    selector: "typeLike",
+    format: ["PascalCase"],
+  },
+  {
+    selector: "variable",
+    modifiers: ["const", "global"],
+    format: ["UPPER_CASE"],
+  },
+  {
+    selector: "variable",
+    modifiers: ["const", "global"],
+    format: ["camelCase"],
+    types: ["function"],
+  },
+];
+
+export default tseslint.config([
+  {
+    ignores: ["node_modules/**/*", ".yarn/**/*", "proto/**", ".next/**"],
+  },
+  // Next config needs compatibility layer for new ESLint versions
+  compat.config({
+    extends: ["next"],
+    settings: {
+      next: {
+        rootDir: ".",
+      },
+    },
+  }),
+  eslint.configs.recommended,
+  tseslint.configs.strict,
+  {
+    languageOptions: {
+      parser: eslintParser,
+      parserOptions: {
+        ecmaVersion: 2021,
```

Comment:
```suggestion
        ecmaVersion: 'latest',
```

Might be time to get with the future? This is quite out of date now.

Reply:
Sure, let's try it 👍